docs/fsharp/language-reference/byrefs.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Byrefs (F#)
+description: Learn about byref and byref-like types in F#, which are used for low-level programming.
+ms.date: 09/02/2018
+---
+
+# Byrefs
+
+F# has two major concepts that deal in the space of low-level programming:
+
+* The `byref`/`inref`/`outref` types, which are a managed pointers. They have restrictions on usage so that you cannot compile a program that would be invalid at runtime.
+* A `byref`-like struct, which is a [structure](structures.md) that has similar semantics and the same compile-time restrictions as `byref<'T>`. One example is <xref:System.Span%601>.
+
+## Syntax
+
+```fsharp
+// Byref types as parameters
+let f (x: byref<'T>) = ()
+let g (x: inref<'T>) = ()
+let h (x: outref<'T>) = ()
+
+// Calling a function with a byref parameter
+let mutable x = 3
+f &x
+
+// Declaring a byref-like struct
+open System.Runtime.CompilerServices
+
+[<Struct; IsByRefLike>]
+type S(count1: int, count2: int) =
+    member x.Count1 = count1
+    member x.Count2 = count2
+```
+
+## Byref, inref, and outref
+
+There are three forms of `byref`:
+
+* `inref<'T>`, a managed pointer for reading the underlying value.
+* `outref<'T>`, a managed pointer for writing to the underlying value.
+* `byref<'T>`, a managed pointer for reading and writing the underlying value.
+
+A `byref<'T>` can be passed anywhere where an `inref<'T>` is expected. Similarly, a `byref<'T>` can be passed anywhere where an `outref<'T>` is expected.
+
+## Using byrefs
+
+To use a `inref<'T>`, all you need is to get a pointer value with `&`:
+
+```fsharp
+open System
+
+let f (dt: inref<DateTime>) =
+    printfn "Now: %s" (dt.ToString())
+
+let dt = DateTime.Now
+f &dt // Pass a pointer to 'dt'
+```
+
+If you wish to write to the pointer by using an `outref<'T>` or `byref<'T>`, then you need to also make the value you grab a pointer to mutable.
+
+```fsharp
+open System
+
+let f (dt: byref<DateTime>) =
+    printfn "Now: %s" (dt.ToString())
+    dt <- DateTime.Now
+
+// Make 'dt' mutable
+let mutable dt = DateTime.Now
+
+// Now you can pass the pointer to 'dt'
+f &dt
+```
+
+If all you are doing is writing the pointer instead of also reading it, consider using `outref<'T>` instead of `byref<'T>`.
```

Comment:
If all you are doing is writing --> If you are only writing
also reading it --> reading it